File: goapi/randtest.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	repeats := 10
	samples := 1000000

	var times [10]int
	var sum int = 0

	for i := 0; i < repeats; i++ {
		start := time.Now().UnixNano() / 1E6
		for j := 0; j < samples; j++ {
			rand.Float64()
		}
		end := time.Now().UnixNano() / 1E6
		times[i] = int(end - start)
		sum += times[i]
	}
	fmt.Printf("Mean time per iteration (ms): %e\n", float64(sum)/float64(repeats*samples))
}

File: pythonapi/randtest.py
from datetime import datetime, timedelta
import random

times = []
repeats = 2
samples = 10000000

for i in range(0,repeats):
    start = datetime.now()
    for j in range(0,samples):
        random.random()
    end = datetime.now()
    dt = (end - start).total_seconds() * 1000
    times.append(dt)

rand_fn_dt_ms = sum(times)/(repeats*samples)
print("Mean time per iteration (ms): ", rand_fn_dt_ms)